src/optimizer/filter_push_down_optimizer.cpp

```diff
@@ -139,6 +140,26 @@ static ColumnPredicateSet getPropertyPredicateSet(const Expression& property,
     return propertyPredicateSet;
 }
 
+// TODO need determine whether the expression is a constant based on its output type
```

Comment:
I don't fully understand this TODO. I thought the expression output data type is guaranteed to be the same as index data type. What do we need to determine here?